X3GWriter.py

```diff
@@ -29,46 +29,55 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.TextMode):
             temp_gcode = tempfile.NamedTemporaryFile("w", delete=False)
             UM.PluginRegistry.PluginRegistry.getInstance().getPluginObject("GCodeWriter").write(temp_gcode, None)
             temp_gcode.close()
-            temp_cfg = None
-            try:
-                temp_cfg = tempfile.NamedTemporaryFile("w", delete=False)
-                self.write_cfg(temp_cfg)
-                temp_cfg.close()
-                temp_x3g = None
+            temp_cfg_name = None
+            machine = self.gpx_machine()
+
+            if machine is None:
                 try:
-                    temp_x3g = tempfile.NamedTemporaryFile("r", delete=False)
-                    temp_x3g.close()
-                    command = self.gpx_command(temp_cfg.name, temp_gcode.name, temp_x3g.name)
-                    try:
-                        process = subprocess.Popen(command)
-                        process.wait() #Wait until it's done converting.
-                        output = process.communicate(b"y")
-                        Logger.log("d", str(output))
-                    except EnvironmentError as e:
-                        Logger.log("e", "System call to X3G converter application failed: {error_msg}".format(error_msg=str(e)))
-                        os.remove(temp_x3g.name)
+                    temp_cfg = tempfile.NamedTemporaryFile("w", delete=False)
+                    temp_cfg_name = temp_cfg.name
+                    self.write_cfg(temp_cfg)
+                    temp_cfg.close()
+                except EnvironmentError as e:
+                    if temp_cfg:
+                        Logger.log("e", "Error writing temporary configuration file {temp_cfg}: {error_msg}".format(temp_cfg=temp_cfg, error_msg=str(e)))
                         os.remove(temp_cfg.name)
-                        os.remove(temp_gcode.name)
-                        return False
-                    #Read from the temporary X3G file and put it in the stream.
-                    stream.write(open(temp_x3g.name, "rb").read())
+                    else: #The NamedTemporaryFile constructor failed.
+                        Logger.log("e", "Error creating temporary configuration file: {error_msg}".format(error_msg=str(e)))
+                    os.remove(temp_gcode.name)
```

Comment:
Out of curiosity, why was the code written to delete the temporary files on error, but keep them on success?

Seems to me the opposite would be more useful when troubleshooting...

Reply:
The intent was to always delete the temporary files, not just on error.